scripts/find_ood.go

```diff
@@ -0,0 +1,453 @@
+			fmt.Println(comID[0])
+		}
+		commitIDs = append(commitIDs, comID[0])
+	}
+
+	return commitIDs
+}
+
+func FindTestCommitDates(path string) map[string]string {
+
+	commitIDs := RetreiveCommitIDs(path)
+
+	// keep the user up to date...
+	fmt.Println("Just found", len(commitIDs), "commits on the atomics repo (since 2022-01-01)... you're in for a treat!")
+	wait := fmt.Sprintf("%s %d %s", "If I had to guess, this will probably take...", (len(commitIDs)/1000)+1, "minutes?")
+	fmt.Println(wait)
+	fmt.Println("Also, FYI, you might get some file errors later on... I swear its not my fault! (okay, it might be)")
+
+	if len(commitIDs) <= 0 {
+		panic("Seems like the path to your atomic-validation-criteria directory is not the default. Try the -criterapath flag!")
+	}
+
+	findTest := regexp.MustCompile(`atomics\/(T\d{4}(?:\.\d{3})?)`)
+	findDate := regexp.MustCompile(`"\w{3}\s+\w{3}\s+\d{1,2}\s+\d{2}:\d{2}:\d{2}\s+\d{4}\s+[+-]\d{4}"`)
+
+	uptoDateMap := make(map[string]string)
+
+	for _, comID := range commitIDs {
+
+		var date string
+
+		if gVerbose {
+			fmt.Println("Getting Date information for commit", comID)
+		}
+
+		comInfo := exec.Command("git", "show", "--name-only", "--format=\"%ad\"", comID)
+
+		comInfo.Dir = filepath.FromSlash(path)
+
+		output, err := comInfo.Output()
+
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		strOutput := string(output)
+
+		// fmt.Println(strOutput)
+
+		//first, find the date of the commit:
+		match := findDate.FindString(strOutput)
+
+		if match != "" {
+			date = match[1 : len(match)-1]
+			if gVerbose {
+				fmt.Println("Date:", date)
+			}
+		} else {
+			if gVerbose {
+				fmt.Println("No date found in the string.")
+			}
+		}
+
+		matches := findTest.FindAllStringSubmatch(strOutput, -1)
+
+		for _, match := range matches {
+
+			if len(match) > 1 {
+				test := match[1]
+				if gVerbose {
+					fmt.Println("Test:", test)
+				}
+
+				//because tests are processed by chronological order, the first time a test is referenced should be its most up to date change
+				if len(uptoDateMap[test]) <= 0 {
+					uptoDateMap[test] = date
+				}
+			}
+		}
+	}
+
+	return uptoDateMap
+}
+
+// unfortunately, the criteria files are special in which they contain ranges of values. This will be treated accordingly.
+func FindCriteriaCommitDates(criteriaPath string) map[string]string {
+
+	//fill in default path
+	if len(criteriaPath) == 0 {
+		criteriaPath = "../atomic-validation-criteria"
+	}
+
+	commitIDs := RetreiveCommitIDs(criteriaPath)
+
+	if len(commitIDs) <= 0 {
+		panic("Seems like the path to your atomic-validation-criteria directory is not the default. Try the -criterapath flag!")
+	}
+
+	findPath := regexp.MustCompile(`\S+\.csv`)
+	findDate := regexp.MustCompile(`"\w{3}\s+\w{3}\s+\d{1,2}\s+\d{2}:\d{2}:\d{2}\s+\d{4}\s+[+-]\d{4}"`)
+
+	criteriaDateMap := make(map[string]string)
+
+	// from the commit IDs, we will get the date, and the path to the file
+	for _, comID := range commitIDs {
```

Comment:
especially since you can use those dates directly in `git log`
```sh
amscwx@ubuntu:~/atomic-red-team$ git log --since "2023-06-05 13:55:37 -0500" atomics
```

Reply:
Wow, if only I knew earlier! I guess you're the first person to discover this because when I searched how to find a commit date this was the only solution I found... That makes this a whole lot faster and a whole lot easier!